joomla/joomla-cms
# [4.0] Add Before and After texts to Number field

The Number field is a very commonly used field in various Joomla settings. It seems to me that this field really lacks one important element: the ability to write a unit of measurement (but not only that).

If you look at another field - Text. This field has the option to add Before and After texts. 
This PR implements the same functionality for Number fields. 
So now you can easily add (minutes, seconds, symbols, px, etc.)

### What is it for?
If you look where this field is used now, it becomes obvious that these features were sorely lacking:
![field-number-occurs](https://user-images.githubusercontent.com/1245155/103775943-d0224280-503f-11eb-918b-0a39ae5e3e31.png)


### Real use example
And here is an example of how I use this field in my RAXO news module.
To me, it looks good and informative, so I decided to share.
![raxo-module-image-settings](https://user-images.githubusercontent.com/1245155/103775922-c8629e00-503f-11eb-951e-4e00f11a770b.png)


### Testing Instructions
Open any manifest file where the Number field is used and try to add new parameters:
```
addonBefore="text" 
addonAfter="text"
```

For example, you can check how it works in the standard Articles - Category module:
/modules/mod_articles_category/mod_articles_category.xml 

```
	<field
			name="introtext_limit"
			type="number"
			label="MOD_ARTICLES_CATEGORY_FIELD_INTROTEXTLIMIT_LABEL"
			default="100"
			filter="integer"
			showon="show_introtext:1"
			addonAfter="characters"
		/>
```
